pkg/api/v2/reporter/reporter.go

```diff
@@ -30,9 +30,63 @@ func (e ResourceReports) Accept(res ...resources.InputResource) ResourceReports
 	return e
 }
 
+// Merge merges the given resourceReports into this resourceReports.
+// Any resources which appear in both resourceReports will
+// have their warnings and errors merged.
+// Errors appearing in both reports, as determined by the error strings,
+// will not be duplicated in the resulting merged report.
 func (e ResourceReports) Merge(resErrs ResourceReports) {
 	for k, v := range resErrs {
-		e[k] = v
+		if firstReport, exists := e[k]; exists {
+			// report already exists for this resource,
+			// merge new report into existing report:
+			secondReport := v
+
+			// Merge warnings lists
+			allWarnings := make(map[string]bool)
+			for _, warning := range firstReport.Warnings {
+				allWarnings[warning] = true
+			}
+			for _, warning := range secondReport.Warnings {
+				if _, found := allWarnings[warning]; !found {
+					firstReport.Warnings = append(firstReport.Warnings, warning)
+				}
+			}
+
+			if firstReport.Errors == nil {
+				// Only 2nd has errs
+				firstReport.Errors = secondReport.Errors
+				e[k] = firstReport
+				continue
+			} else if secondReport.Errors == nil {
+				// Only 1st has errs
+				continue
+			}
+
+			// Both first and second have errors for the same resource:
+			if errs1, isMulti := firstReport.Errors.(*multierror.Error); isMulti {
+				if errs2, isMulti := secondReport.Errors.(*multierror.Error); isMulti {
+					// Any errors which are identical won't be duplicated,
+					// Any errors which are unique will be added to the final list
+					allErrsMap := make(map[string]error)
+					for _, err := range errs1.Errors {
+						allErrsMap[err.Error()] = err
+					}
+					for _, err := range errs2.Errors {
+						if _, found := allErrsMap[err.Error()]; !found {
+							allErrsMap[err.Error()] = err
+							errs1.Errors = append(errs1.Errors, err)
+						}
+					}
+				}
+				firstReport.Errors = errs1
```

Comment:
what if `errs1` is multi but `errs2` is not?

Reply:
and vice-versa

Reply:
As far as I'm aware these should always be multierrs or nil... but if either (or both) of them is a non-nil non-multi err, what would be the expected behavior here? Concat the error strings?

Reply:
I guess if one is multi and the other is not, we can add the non-multi to the multi... and if both are just regular errors then we concat?

Reply:
if they both aren't multis, we could just make a multi